### PR TITLE
Always return execution data inline when no remote data provider set

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
When no remote data provider is set or the local/noop option is specified, always return execution inputs and outputs inline for GetData requests

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/900

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/runtime/application_config_provider.go [hunk 1]
```diff
@@ -21,7 +23,13 @@ const externalEvents = "externalEvents"
 var databaseConfig = config.MustRegisterSection(database, &interfaces.DbConfigSection{})
 var flyteAdminConfig = config.MustRegisterSection(flyteAdmin, &interfaces.ApplicationConfig{})
 var schedulerConfig = config.MustRegisterSection(scheduler, &interfaces.SchedulerConfig{})
```
Comment (EngHabu) [inline pkg/runtime/application_config_provider.go:25]: Do you know why are we registering many sections in one package? it works... but not needed I guess....

File: pkg/runtime/application_config_provider.go [hunk 2]
```diff
@@ -21,7 +23,13 @@ const externalEvents = "externalEvents"
...
-var remoteDataConfig = config.MustRegisterSection(remoteData, &interfaces.RemoteDataConfig{})
+
+var defaultRemoteDataConfig = &interfaces.RemoteDataConfig{
+	Scheme:         common.Local,
+	MaxSizeInBytes: 1048576, // 1 Mib
+}
+var remoteDataConfig = config.MustRegisterSection(remoteData, defaultRemoteDataConfig)
```
Comment (EngHabu) [inline pkg/runtime/application_config_provider.go:31]: nit: Shouldn't we also generate pflags provider for these?

**Review by EngHabu — DISMISSED** [review]

File: pkg/manager/impl/execution_manager.go [hunk 3]
```diff
@@ -1111,15 +1111,16 @@ func (m *ExecutionManager) GetExecutionData(
 		Inputs:  &inputsURLBlob,
 	}
 	maxDataSize := m.config.ApplicationConfiguration().GetRemoteDataConfig().MaxSizeInBytes
-	if maxDataSize == 0 || inputsURLBlob.Bytes < maxDataSize {
+	remoteDataScheme := m.config.ApplicationConfiguration().GetRemoteDataConfig().Scheme
+	if remoteDataScheme == common.Local || inputsURLBlob.Bytes < maxDataSize {
```
Comment (kumare3) [inline pkg/manager/impl/execution_manager.go:1115]: what if we change it to always return inline and sometimes return the signed ulr?

File: pkg/runtime/application_config_provider.go (hunk 1 above)
Comment (katrogan) [reply pkg/runtime/application_config_provider.go:25]: oh just as a logical/organizational breakdown

File: pkg/manager/impl/execution_manager.go (hunk 3 above)
Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1115]: then we should just get rid of the signed url altogether, no?

Comment (kumare3) [reply pkg/manager/impl/execution_manager.go:1115]: Thats what I want to do, but I dont know if that will work for all cases at Lyft? If we have a metric lets collect that?

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:1115]: we have a metric for observing the bytes size - can follow up to find out what it trends towards

**Review by EngHabu — APPROVED** [review]

**Pull request merged**